classes/04class/exercises/c04-iac02/frdvo/vpc.tf

```diff
@@ -0,0 +1,123 @@
+# VPC devopsacademy_iac
+resource "aws_vpc" "devopsacademy_iac_vpc" {
+  cidr_block = var.vpc_cidr
+
+  tags = {
+    Name = "devopsacademy_iac"
+  }
+}
+
+# 4 Subnets
+resource "aws_subnet" "public_subnet_1" {
+  vpc_id            = aws_vpc.devopsacademy_iac_vpc.id
+  cidr_block        = var.subnets.public_subnet_1.cidr
+  
+  tags = {
+    Name = var.subnets.public_subnet_1.name
+  }
+}
```

Comment:
Since that your variable if of type map, we could use `for_each` to avoid duplicate the subnet resource block:

```diff
-resource "aws_subnet" "public_subnet_1" {
+resource "aws_subnet" "public_subnet" {
+    for_each          = var.subnets.public_subnet_1
     vpc_id            = aws_vpc.devopsacademy_iac_vpc.id
-    cidr_block        = var.subnets.public_subnet_1.cidr
+    cidr_block        = each.value.cidr

  tags = {
-     Name = var.subnets.public_subnet_1.name
+    Name = each.key
  }
}
```

```hcl
# where
each.key = public_subnet_1
each.value = { 
    name = "public_subnet_1",
    cidr = "10.0.101.0/24"
}
```

> since that the `each.key` would be the same as `name`, you could remove the field `name`.

Reply:
I like the idea, I will try to implement. Thanks for this tip!